gnomad/utils/file_utils.py

```diff
@@ -6,15 +6,69 @@
 import os
 import subprocess
 import uuid
-from typing import List, Optional, Tuple, Union
+from concurrent.futures import ThreadPoolExecutor
+from typing import List, Dict, Optional, Tuple, Union
 
 import hail as hl
+from hailtop.aiotools import LocalAsyncFS, RouterAsyncFS, AsyncFS
+from hailtop.aiogoogle import GoogleStorageAsyncFS
+from hailtop.utils import bounded_gather, tqdm
 
 logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
 
+async def parallel_file_exists(fnames: List[str]) -> Dict[str, bool]:
+    """
+    Check whether a large number of files exist.
+
+    Created for use with hail Batch jobs.
+    Normal `file_exists` function is very slow when checking a large number of files.
+
+    :param fnames: List of file names to check.
+    :return: Dictionary of file names (str) and whether the file exists (boolean).
+    """
+
+    async def low_level_async_file_exists(fs: AsyncFS, url: str):
```

Comment:
Is there any reason to add documentation to this? I know it is an inner function so I don't think it is needed, but we have typically added them in the past. I'm fine either way